### PR TITLE
Map export

### PR DESCRIPTION
**Pull request by titilambert**

### CODE REVIEW

File: tml/constants.py [hunk 1]
```diff
@@ -75,3 +75,4 @@
 }
 
 SPEEDUPINDEX = 28
+R_TILEINDEX = dict([(v, k) for k, v in TILEINDEX.items()])
```
Comment (timakro) [inline tml/constants.py:78]: Because the list is not exhaustive maybe we should just always get the image.

File: tml/datafile.py [hunk 2]
```diff
@@ -107,10 +107,6 @@ def __init__(self, map_path):
                     license = decompress(self.get_compressed_data(f, license))[:-1]
                 else:
                     license = None
-                if settings > -1:
-                    settings = decompress(self.get_compressed_data(f, settings)).split(b'\x00')[:-1]
-                else:
-                    settings = None
```
Comment (timakro) [inline tml/datafile.py:113]: Why those changes?

**Review by timakro — COMMENTED** [review]
Thanks for your png export feature!

I just want to cherry-pick that comment. The mapinfo settings feature should stay, there is already support for the front layer and the other 3 scripts seem too DDNet specific.

And what's the matter with all the mapres image changes?

File: tml/datafile.py (hunk 2 above)
Comment (titilambert) [reply tml/datafile.py:113]: I don't know, but
- DDnet removed it from their fork
- With teeworlds 0.7.2 maps it's doesn't work (if those lines are still there)

**Comment by titilambert** [comment]
I just updated the mapres with the latest mapres (teeworlds 0.7.2). If I'm not doing it the rendering (for the map) become really weird...

File: tml/constants.py (hunk 1 above)
Comment (titilambert) [reply tml/constants.py:78]: Which image are you talking about ?

**Comment by titilambert** [comment]
So today I'm able to do:
```
    map_file_path = 'dm1.map'
    map_name = os.path.basename(map_file_path).rsplit(".", 1)[0]
    if not os.path.exists('screenshots'):
        os.makedirs('screenshots')
    output_file = 'screenshots/' + map_name + '.png'

    t = Teemap(map_file_path)
    t.export_to_png(output_file, False)
```

And I get `screenshots/dm1.png` created.
I also tested with some non offcial maps and it's working really well ! The only issue is the performances. If the map is huge it can't take more than 15 minutes ...

**Comment by timakro** [comment]
I cherry-picked [your commit](https://github.com/timakro/tml/commit/f2c9625bd378845105b79471ee9a508a9afeb7ef), Thanks!

I made the following changes:
* Major performance improvement by loading the images just once and keeping them in memory instead of loading them again for each tile
* The library can load maps with and without mapinfo settings now, so 0.6 and 0.7 maps work
* The `mapres` folder stays like it was before, instead I added a `mapres_0.7` directory with the new mapres
* There is a new `render_gamelayer` parameter which defaults to off

This is how you would render a 0.7 map with the game layer.

```python
from tml.tml import Teemap
t = Teemap('dm1.map', game_version='0.7')
t.export_to_png('dm1.png', render_gamelayer=True)
```

**Pull request closed without merging**